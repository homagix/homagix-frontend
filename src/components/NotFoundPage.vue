<script setup lang="ts"></script>

<template>
  <h2>Hast du dich verirrt?</h2>
  <p>Die angeforderte Seite existiert leider nicht.</p>
  <router-link to="/">Bring mich in Sicherheit!</router-link>
</template>
