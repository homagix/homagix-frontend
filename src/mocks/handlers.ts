import { rest } from "msw"

export const handlers = [
  rest.get("http://localhost:8200/api/dishes", (req, res, ctx) => {
    return res(ctx.json({ dishes: [{ id: 1, name: "Pizza" }]}))
  }),
]
