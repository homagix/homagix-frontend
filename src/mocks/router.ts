import { defineComponent } from "vue"
import { createWebHistory } from "vue-router"
import { createRouter } from "vue-router"

export function createRouterMock(component: ReturnType<typeof defineComponent>, extraPaths: string[] = []) {
  const history = createWebHistory()
  const routes = ["/", ...extraPaths].map(path => ({ path, component }))
  return createRouter({ history, routes })
}
