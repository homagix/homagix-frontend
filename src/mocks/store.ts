import { vi } from "vitest"
import { createStore } from "vuex"

export function createStoreMock(state: object = {}, options: { actions?: string[]; mutations?: string[] } = {}) {
  function createMocksRecord(names?: string[]) {
    if (!names) {
      return undefined
    }
    return Object.assign({}, ...names.map(name => ({ [name]: vi.fn() })))
  }

  const actionMocks = createMocksRecord(options.actions)
  const mutationMocks = createMocksRecord(options.mutations)
  const store = createStore({
    state() {
      return state
    },
    actions: actionMocks,
    mutations: mutationMocks,
  })
  return { store, actionMocks, mutationMocks }
}
