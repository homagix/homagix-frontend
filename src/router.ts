import { createRouter, createWebHistory } from "vue-router"
import RecipesList from "@/components/RecipesList.vue"

export default createRouter({
  history: createWebHistory(import.meta.env.BASE_URL),
  routes: [
    { name: "home", path: "/", redirect: "/recipes" },
    { name: "recipes", path: "/recipes", component: RecipesList },
  ],
})
