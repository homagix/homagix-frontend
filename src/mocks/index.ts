import { mount, shallowMount } from "@vue/test-utils"
import { createRouterMock } from "./router"
import { createStoreMock, StoreMockOptions } from "./store"
import Oruga from "@oruga-ui/oruga-next"

type MountOptions = {
  shallow?: boolean
  props?: Record<string, any>
  state?: object
  storeOptions?: StoreMockOptions
  extraPaths?: string[]
}

export function mountComponent<C>(component: C, options: MountOptions) {
  const { store, actionMocks, mutationMocks } = createStoreMock(options.state || {}, options.storeOptions || {})
  const router = createRouterMock(component, options.extraPaths || [])
  const m = options.shallow === undefined || options.shallow ? shallowMount : mount
  const wrapper = m(component, { props: options.props || {}, global: { plugins: [store, router, Oruga] } })

  return { wrapper, store, router, actionMocks, mutationMocks }
}
