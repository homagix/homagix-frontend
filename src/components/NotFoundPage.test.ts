import { mount } from "@vue/test-utils"
import { describe, expect, it, vi } from "vitest"
import { createRouterMock } from "../mocks/router"
import NotFoundPage from "@/components/NotFoundPage.vue"
import { createStore } from "vuex"

describe("NotFoundPage", () => {
  it("should render", () => {
    const router = createRouterMock(NotFoundPage)
    const setUserMutation = vi.fn()
    const store = createStore({
      state() {
        return {}
      },
      mutations: { SET_USER: setUserMutation },
    })
    const wrapper = mount(NotFoundPage, {
      global: { plugins: [store, router] },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
