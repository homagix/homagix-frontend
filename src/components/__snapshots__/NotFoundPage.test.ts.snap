// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFoundPage > should render 1`] = `
"<div class="title is-4">Hast du dich verirrt?</div>
<p>Die angeforderte Seite existiert leider nicht.</p>
<a href="/" class="">Bring mich in Sicherheit!</a>"
`;
